File: CovidTracerApp/app/src/main/java/com/example/covidtracerapp/database/ContactedEntity.java
package com.example.covidtracerapp.database;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity(tableName = "contacted")
public class ContactedEntity {

    @PrimaryKey(autoGenerate = true)
    private int  id;
    private Date    contactedAt;
    @Ignore
    public ContactedEntity(int id, Date contactedAt) {
        this.id = id;
        this.contactedAt = contactedAt;
    }
    public ContactedEntity(Date contactedAt) {
        this.contactedAt = contactedAt;
    }
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getContactedAt() {
        return contactedAt;
    }

    public void setContactedAt(Date contactedAt) {
        this.contactedAt = contactedAt;
    }
}

File: CovidTracerApp/app/src/main/java/com/example/covidtracerapp/database/ContactedEntity.kt
package com.example.covidtracerapp.database

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import java.util.*
import kotlinx.android.parcel.Parcelize

const val ContactedTable = "contacted"
@Entity(tableName = ContactedTable)
data class ContactedEntity(@PrimaryKey val id: String, val contactedAt: Date)